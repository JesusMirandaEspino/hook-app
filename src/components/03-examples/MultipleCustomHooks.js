import React from 'react'
import '../../components/02-useEffect/effects.css';
import {  Usefetch } from '../../Hooks/Usefetch';

export const MultipleCustomHooks = () => {

    const { loading, data } =  Usefetch( 'https://www.breakingbadapi.com/api/quotes/1' );
    const { author, quote } = !!data && data[0];


    return (
        <div>
            <h1>BreakingBag Quotes </h1>

            <hr/>

            {
                loading 
                ?
                (
                    <div className="alert alert-info text-center" >
                            <p>Loading...</p>
                    </div>
                )
                :
                (
                    <blockquote  className="blockquote text-right">
                        <p className="mb-0" > { quote }   </p>
                        <footer className="blockquote-footer mt-2" >{ author }</footer>
                    </blockquote>
                )
            }
        </div>

    )
}
