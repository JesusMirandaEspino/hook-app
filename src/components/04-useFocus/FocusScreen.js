import React, { useRef } from 'react'
import '../../components/02-useEffect/effects.css';


export const FocusScreen = () => {

    const inputRef = useRef();


    const handleClick  = () => {
        inputRef.current.select();
    }

    return (

        <div>

            <h1>Use focus</h1>

            <hr/>


            <label  for="" className="form-control " > Su nombre </label>
            <input ref={ inputRef } className="form-control " />


            <button className="btn btn-outline-primary mt-4"  onClick={handleClick} >Focus</button>


        </div>

    );


}





