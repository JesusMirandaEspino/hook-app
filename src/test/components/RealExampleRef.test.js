import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import React from 'react';
import { RealExampleRef } from '../../components/04-useFocus/RealExampleRef';
import { MultipleCustomHooks } from '../../components/03-examples/MultipleCustomHooks';

describe( 'Pruebas con </RealExampleRef >', () => {

    const wrappers = shallow( <RealExampleRef /> );

    test( 'Debe mostrarse correctamente', () => {
        expect( wrappers ).toMatchSnapshot();
        expect( wrappers.find( 'MultipleCustomHooks' ).exists() ).toBe(false);

    });


    test( 'Debe mostras correctamente <MultipleCustomHooks />', () => {

        wrappers.find( 'button' ).simulate('click');
        expect( wrappers.find( 'MultipleCustomHooks' ).exists() ).toBe(true);

    });


});