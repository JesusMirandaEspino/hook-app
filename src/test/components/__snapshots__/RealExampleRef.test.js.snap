// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas con </RealExampleRef > Debe mostrarse correctamente 1`] = `
<dvi>
  <h1>
    Real Example Ref
  </h1>
  <hr />
  <button
    className="btn btn-primary mt-5"
    onClick={[Function]}
  >
     Show/Hide  
  </button>
</dvi>
`;
